tests/e2e_all/dockerfiles/Dockerfile.c.branch

```diff
@@ -0,0 +1,58 @@
+FROM atsigncompany/cbuildimage:CMake-3.31.4@sha256:0b443da3bccd746e23323eb374d19ba156b497ecbee6c5e84f97d77954515cbf AS buildimage
+
+ENV URL=https://github.com/atsign-foundation/noports.git
+ENV REPO_DIR=/app/repo
+ENV OUTPUT_DIR=/usr/local/bin
+
+ARG branch=trunk
+
+RUN set -eux ; \
+    mkdir -p ${REPO_DIR} ${OUTPUT_DIR} ; \
+    apt-get update ; \
+    apt-get install -y git gcc ; \
+    cd ${REPO_DIR} ; \
+    git clone ${URL} . ; \
+    git checkout ${branch} ; \
+    cd ${REPO_DIR}/packages/c/sshnpd ; \
+    cmake -S . -B build -DCMAKE_C_COMPILER=gcc -DBUILD_SHARED_LIBS=off -DCMAKE_C_FLAGS="-Wno-error -pthread" ; \
+    cmake --build build ; \
+    mv ./build/at_activate ${OUTPUT_DIR}/at_activate ; \
+    mv ./build/sshnpd ${OUTPUT_DIR}/sshnpd ; \
+    mv ./build/libsshnpd-lib.a ${OUTPUT_DIR}/libsshnpd-lib.a ; \
+    cd ${REPO_DIR}/packages/c/srv ; \
+    cmake -S . -B build -DCMAKE_C_COMPILER=gcc -DATAUTH_BUILD_EXECUTABLES=OFF -DBUILD_SHARED_LIBS=off -DCMAKE_C_FLAGS="-Wno-error -pthread" ; \
+    cmake --build build ; \
+    mv build/srv ${OUTPUT_DIR}/srv
+
+FROM debian:stable-slim@sha256:5484adc33b4c352c5a9f4c4ae295fc49aed1cb54a7a0712a1b29674fb6f4f10f AS runtime
+
+ENV USER=atsign
+ENV HOMEDIR=/${USER}
+ENV USER_ID=1024
+ENV GROUP_ID=1024
+
+COPY --from=buildimage /usr/local/bin/at_activate /usr/local/bin/at_activate
+COPY --from=buildimage /usr/local/bin/sshnpd /usr/local/bin/sshnpd
+COPY --from=buildimage /usr/local/bin/libsshnpd-lib.a /usr/local/bin/libsshnpd-lib.a
+COPY --from=buildimage /usr/local/bin/srv /usr/local/bin/srv
+
+RUN set -eux ; \
+    apt-get update ; \
+    apt-get install -y openssh-server sudo vim nano iproute2 nmap tmux curl cron ; \
+    groupadd --gid ${GROUP_ID} ${USER} ; \
+    useradd --system --shell /bin/bash --home ${HOMEDIR} --uid ${USER_ID} --gid ${GROUP_ID} ${USER} ; \
+    usermod -aG sudo ${USER} ; \
+    mkdir -p ${HOMEDIR}/.ssh ${HOMEDIR}/.atsign/keys ; \
+    chown -R ${USER}:${USER} ${HOMEDIR} ; \
+    touch ${HOMEDIR}/.ssh/authorized_keys ; \
+    ex +"%s/^%sudo.*$/%sudo ALL=(ALL:ALL) NOPASSWD:ALL/g" -scwq! /etc/sudoers ; \
```

Comment:
why a vim ex command instead of sed?

Reply:
This should be the same as sed -e 's/%sudo.*$/%sudo ALL=(ALL:ALL) NOPASSWD:ALL/g' -i /etc/sudoers and you don't need to install vim & the vim runtime (which is heavier than just vi)

Reply:
Done in https://github.com/atsign-foundation/noports/pull/1757/commits/64fcfeef074a17bcbc08ed0a7f3af20597744b74